lib/keystore.hh

```diff
@@ -10,19 +10,23 @@ namespace rpm {
 
 class keystore {
 public:
+    const std::string name;
     virtual rpmRC load_keys(rpmtxn txn, rpmKeyring keyring) = 0;
     virtual rpmRC import_key(rpmtxn txn, rpmPubkey key, int replace = 1, rpmFlags flags = 0) = 0;
     virtual rpmRC delete_key(rpmtxn txn, rpmPubkey key) = 0;
+    virtual rpmRC delete_store(rpmtxn txn) = 0;
 
     virtual ~keystore() = default;
+    keystore(std::string n): name(n) {};
```

Comment:
Something was nagging me about this name thing all along, and now I know what it was:  
We're now duplicating info with getKeystore() here. Instead, the name should be a static property, and then we should have an std::array of available types. And then getKeystore() can use that array to look up the right thing from the classes themselves. Like the rpmdb backend does. And, when you have a function that returns a backend based on a name, both getKeystore() and the --from logic can use that.

Reply:
Actually gcc gives a pretty helpful hint here when trying to add something like `static std::string name = "fs";` to the keystore:
> /home/pmatilai/repos/rpm/lib/keystore.hh:22:30: error: ‘constexpr’ needed for in-class initialization of static data member ‘const std::string rpm::keystore_fs::name’ of non-integral type [-fpermissive]
   22 |     static const std::string name = "fs";

`static constexpr std::string name = "fs";` merrily compiles.

Reply:
Anyway, the getKeystore() approach in the PR is totally okay - the knowledge is contained in exactly one place now.